### ISSUE TITLE
Support aarch64 Linux

### ISSUE THREAD

**Issue by Pokemetti** [ColonelPhantom/lite-xl-tmt#17]
I'm trying to get this module to run on aarch64 Linux right now (NanoPi R6C) and it compiles fine, even gets loaded correctly by lite-xl (no errors in the log), but when trying to open a new tmt-window the whole editor crashes with the message "called newuserdata non-uv" on the console.

**Comment by ColonelPhantom** [comment]
Hi,

I think the error you're experiencing is due to an incompatibility between the plugin and the Lua version your build of Lite XL is using. Can you tell which version of Lua it is using? In theory version 0.4h of the plugin should work with both Lua 5.4 and LuaJIT, but it's been a while since I looked at the code for it.

I don't think the error is related to running on aarch64.

**Comment by Pokemetti** [comment]
Well, I'm running the standard repo version of Lua5.4.4 on Debian, so I guess the problem might be there? I can try to build it myself I guess and look whether that works.

**Comment by Pokemetti** [comment]
I built Lua 5.4.6 and installed it myself, rebuilt lite-xl and it still crashes. I don't even have the repo version of Lua installed, so it can't really be that.

**Comment by ColonelPhantom** [comment]
> I built Lua 5.4.6 and installed it myself, rebuilt lite-xl and it still crashes. I don't even have the repo version of Lua installed, so it can't really be that.

Lite XL is usually statically linked with its own build of Lua, not the repository version. I believe that if you build the editor yourself you can choose to link it against system Lua, although I don't know how the details off the top of my head.

Most likely I'm guessing something is going wrong with Lite XL's native plugin API. Can you confirm what Lua version is used and what Lite XL version is used? (For Lua version, I guess the easiest way to find is out is to look at the value of `_VERSION`.)

**Comment by Pokemetti** [comment]
Nope, the problem was quite simple apparently, I just replaced lua_newuserdata with lua_newuserdatauv with nuvalue set to 0. At least it doesn't crash now and seems to show a working terminal. lua_newuserdata is supposed to be accessable only through a macro, I'm not sure why it doesn't work but I saw some macro declaring it as a symbol in some header? If the library tried to access lua_newuserdata from the library instead of from the header that would explain why it crashed at least. But honestly I barely have any idea how the Lua C API works so I'm happy I got it working somehow.
